### PR TITLE
Discover: Add UI for connection test for Connect My Computer

### PR DESCRIPTION
**Pull request by ravicious**
cmc-conn-test

This PR adds just the UI for a proper connection test that actually attempts to connect to the node and shows diagnostic messages. The diagnostic messages are yet to be customized towards Connect My Computer (#32206), for now the test behaves the same way as if we were testing a regular SSH node.

I also added a feature flag so that I don't have to wait with merging this PR before I have the backend changes ready.

The implementation is done in a few simple steps:

1. Take the existing test step, rename it to `LegacyTestConnection`.
1. Add a new step in its place with the contents from `Server/TestConnection`.
1. Take parts related to Connect My Computer from `LegacyTestConnection` and fit them into the new component copied from Server.

I had to balance [Figma designs](https://www.figma.com/file/uLevdNsEnIvvLDSZ9sQqXM/Discover-Access?type=design&node-id=1386-30152&mode=design&t=6yyuYcsKq3c645DS-4) with how the existing implementation looks like, both for Connect My Computer and a regular server. Mostly this meant keeping step numbers on the same line as descriptions and how particular buttons and actions are called.

To test this locally, you need to enable the feature flag:

```javascript
localStorage.setItem('grv_teleport_discover_connect_my_computer_new_connection_test_enabled', 'true')
```

### CODE REVIEW

File: web/packages/teleport/src/Discover/ConnectMyComputer/TestConnection/TestConnection.tsx [hunk 1]
```diff
@@ -68,168 +84,227 @@ export const TestConnection = (props: AgentStepProps) => {
...
+  function testConnection(login: string, mfaResponse?: MfaAuthnResponse) {
+    return runConnectionDiagnostic(
+      {
+        resourceKind: 'node',
+        resourceName: props.agentMeta.resourceName,
+        sshPrincipal: login,
+      },
+      mfaResponse
+    );
+  }
```
Comment (ravicious) [inline web/packages/teleport/src/Discover/ConnectMyComputer/TestConnection/TestConnection.tsx:141]: In #34952 I mentioned sharing these two functions between Connect My Computer and Server connection tests, but I think for now I'm going to keep them this way. It's not much code, in the end it calls functions from `useConnectionDiagnostic` which is already shared. And to make the diagnostic messages be more relevant to Connect My Computer, I'll have to add some extra parameter to `runConnectionDiagnostic` anyway.

So in the end I don't think there's much to be gained from sharing these two functions.

File: web/packages/teleport/src/Discover/ConnectMyComputer/TestConnection/TestConnection.story.tsx [hunk 2]
```diff
@@ -74,14 +112,22 @@ export const MultipleLogins = () => {
...
+    handlers: {
+      connectMyComputerLogins: [
+        rest.get(cfg.api.connectMyComputerLoginsPath, (req, res, ctx) =>
+          res(
+            ctx.json({
+              logins: [
+                'foo',
+                'bar',
+                'baz',
+                'czesława_maria_de_domo_cieślak_primo_voto_gospodarek_secundo_voto_kowalczyk',
```
Comment (ravicious) [inline web/packages/teleport/src/Discover/ConnectMyComputer/TestConnection/TestConnection.story.tsx:124]: Server/TestConnection uses a shorter login:

https://github.com/gravitational/teleport/blob/170464fb5651f8bdfcaf4a3582bc5575d3a1b52f/web/packages/teleport/src/Discover/Server/TestConnection/TestConnection.story.tsx#L36-L39

and the width of the login select is set to be just wide enough to contain it, but it fails with the full name of Violetta Villas. 😭

login-select-with-long-values

For now I'm just not going to try to fix what's broken as I couldn't quickly figure out how to make react-select work in this scenario.

**Review by zmb3 — APPROVED** [review]

File: web/packages/teleport/src/Discover/ConnectMyComputer/index.ts [hunk 3]
```diff
@@ -31,14 +33,19 @@ export const ConnectMyComputerResource: ResourceViewConfig<ResourceSpec> = {
...
-      // TODO(ravicious): Manually emit success event on test connection success.
-      manuallyEmitSuccessEvent: true,
-    },
+    storageService.isDiscoverConnectMyComputerNewConnectionTestEnabled()
+      ? {
+          title: 'Test Connection',
+          component: TestConnection,
+          eventName: DiscoverEvent.TestConnection,
+          manuallyEmitSuccessEvent: true,
+        }
```
Comment (kimlisa) [inline web/packages/teleport/src/Discover/ConnectMyComputer/index.ts:42]: is this temporary?

**Review by kimlisa — APPROVED** [review]

File: web/packages/teleport/src/Discover/ConnectMyComputer/TestConnection/TestConnection.tsx [hunk 4]
```diff
@@ -72,60 +130,171 @@ export function TestConnection(props: AgentStepProps) {
...
+ * To display the first step, fetchLoginsAttempt must be successful, hence why we show errors from
+ * fetchLoginsAttempt within the first step.
+ */
+const FetchLoginsAttemptError = (props: {
+  children: React.ReactNode;
+  buttonText: string;
+  buttonOnClick: () => void;
+}) => (
+  <StepSkeletonPickUser>
+    <>
```
Comment (gzdunek) [inline web/packages/teleport/src/Discover/ConnectMyComputer/TestConnection/TestConnection.tsx:281]: Unnecessary `<>`.

**Review by gzdunek — APPROVED** [review]

File: web/packages/teleport/src/Discover/ConnectMyComputer/TestConnection/TestConnection.tsx (hunk 4 above)
Comment (ravicious) [reply web/packages/teleport/src/Discover/ConnectMyComputer/TestConnection/TestConnection.tsx]: There was a few more unnecessary fragments in this file. react/jsx-no-useless-fragment doesn't catch that, even with `allowExpressions` disabled.

**Pull request merged**